Check http code when downloading vulnerabilities data

Allda opened the pull request:
When http requests to Red Hat's oval data doesn't return StatusOK it
returns empty vulnerability list with no error. This should not happen.

This commits check the return code of http request to Red Hat's oval data.

This is also related to #596

Allda committed:
Check http code when downloading Red Hat's oval data

When http requests to Red Hat's oval data doesn't return StatusOK it
returns empty vulnerability list with no error. This should not happen.

This commits check return code of http request to Red Hat's oval data.

MackJM commented:
Although this would definitely help with what happened with RHEL. The same issue exists in debian and oracle. I suggest we move this to a package that we can share for all three. That way we can also set the User-Agent for our requests as well.

jzelinskie commented:
There doesn't seem to be much reusable code in this change to pull out into something shareable. However, I would like to eventually create an OVAL library that can be reused.

If you add this check for the few other vulnsrcs, I'll merge it.

Allda committed:
Check http code when downloading Debian's vulnerabilities

When http requests to Debian's vulnerability data doesn't return StatusOK it
returns empty vulnerability list with no error. This should not happen.

This commits check return code of http request to Debian's
vulnerabilities data.

Allda committed:
Check http code when downloading Oracle's oval data

When http requests to Oracle's oval data doesn't return StatusOK it
returns empty vulnerability list with no error. This should not happen.

This commits check return code of http request to Oracle's oval data.

Allda commented:
@jzelinskie I added HTTP status check to others vulnsrcs modules.

Allda commented:
Closing this PR since @jzelinskie added new http module which already does the status code check.
